ext/mbstring/tests/utf_encodings.phpt

```diff
@@ -0,0 +1,330 @@
+--TEST--
+Torture test for UTF-{7,8,16,32}
+--SKIPIF--
+<?php
+extension_loaded('mbstring') or die('skip mbstring not available');
+if (getenv("SKIP_SLOW_TESTS")) die("skip slow test");
+?>
+--FILE--
+<?php
+srand(232); /* Make results consistent */
+mb_substitute_character(0x25); // '%'
+include('encoding_tests.inc');
+
+// in UTF-32BE
+$validCodepoints = array();
+$fp = fopen(realpath(__DIR__ . '/data/UnicodeData.txt'), 'r+');
```

Comment:
Why do we need UnicodeData here? The file is nearly 2MB large, so I would not want to bundle it. We generally only use UnicodeData.txt to generate own tables.

For the use-case here, I think it would be sufficient to specify some broad ranges of characters to check, if we want to avoid checking wide swathes of unassigned codepoints.

Reply:
Sure. That will do.